feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.spec.ts

```diff
@@ -249,66 +270,84 @@ describe('ConfigUtilsService', () => {
         values: [],
       };
 
-      it('should delegate to keyboard focus service with attribute', () => {
-        spyOn(windowRef, 'isBrowser').and.returnValue(true);
-        const focusedElements = createFocusedElements('ATTR', 2, 3);
+      function spyFocus(): HTMLElement | undefined {
+        const selectedValue = attribute.values?.find((value) => value.selected);
+        if (selectedValue) {
+          const valueUiKey = attribute.name + '--' + selectedValue.valueCode;
+          const focusedElement = focusedElements.find(
+            (el) => el.id.indexOf(valueUiKey) !== -1
+          );
+          if (focusedElement) {
+            spyOn(focusedElement, 'focus').and.callThrough();
+          }
+          return focusedElement;
+        }
+        return undefined;
+      }
+
+      beforeEach(() => {
+        focusedElements = fixture.debugElement
+          .queryAll(By.css('label'))
+          .map((el) => el.nativeNode);
+
         document.querySelector = jasmine
           .createSpy('HTML Element')
           .and.returnValue(focusedElements);
+
         spyOn(keyboardFocusService, 'findFocusable').and.returnValue(
           focusedElements
         );
+      });
 
+      it('should not set focus because attribute does not contain any values', () => {
+        spyOn(windowRef, 'isBrowser').and.returnValue(true);
+        const focusedElement = spyFocus();
         classUnderTest.focusValue(attribute);
         expect(keyboardFocusService.findFocusable).toHaveBeenCalledTimes(1);
+        if (focusedElement) {
+          expect(focusedElement.focus).toHaveBeenCalled();
+        }
```

Comment:
based on the test name i would have expect a `expect(focusedElement.focus).not.toHaveBeenCalled()`